### DISCUSSION TITLE
local_files_only with safetensors required config

### DISCUSSION THREAD

**Discussion by olegchomp** [huggingface/diffusers#8544] [Q&A] [answered]
Hi! I'm trying to use StableDiffusionPipeline.from_single_file and local_files_only with safetensors model. Depend on selected model (from civitai) it allways download different configs (text_encoder, scheduler, model_index, tokenizer and etc.). 
1) In case if PC doesn't have internet it become to imposible switch model if it's config not exists. 
2) Can this config be unifited between for ex. all SD 1.5 models? 
3) How to point pipeline to this files?
4) If not, how to change folder for this configs? Because it allways download in folder where script was executed what is not good at all.

**Comment by olegchomp** [comment]
Found multiple issues like https://github.com/huggingface/diffusers/discussions/6836 and https://github.com/huggingface/diffusers/issues/4021, but can't get idea, do i need those 10 gb models? Because if set to not use local files it only download around 20-30 mb only.

**Reply by asomoza** [reply]
Hi, you don't need the safetensor, bin or onnx files, you only need the json and txt files (not sure if all of them).

Also each architecture needs only one config file, so one config for SD 1.5, SDXL and so on.

**Reply by olegchomp** [reply]
Thank you!
So i've tried with placing folder with those files in same root as execution script. But how i can change this? For ex. to load it from cache folder like models/config instead of root directory.
I've tried like in https://github.com/huggingface/diffusers/discussions/6836 but in that way it required to download model.

upd. followed this https://github.com/huggingface/diffusers/pull/5440 got download configs. But it still required files to be in same root path as execution script or in default hub cache folder, looks like it ignoring cache_dir if provided.

**Reply by asomoza** [reply]
That's a very special use case and probably we won't add it unless there's a lot users that need this. The logic here is that we try to load the config files from the HF hub, when you're working offline, you'll need to have the same file structure locally as the repo name.

If you don't want to use this, you have other options:

- do first an online run so the config files get downloaded.
- do an online run in other system with internet and copy the cache folder relevant data to the offline one (custom cache dir should work).
- you can just manually load the text encoders (with the weights though) and just pass them as an argument to the pipeline.

The last one but harder one is that you change the `FromSingleFileMixin` to fit your needs but I'm guessing you don't want to do that. 

If it helps, I did an app before and I did all the load from single file myself to fit what I needed using the diffusers code as a guide.

cc @DN6 for visibility and if maybe I'm missing something.

**Reply by olegchomp** [reply]
Thank you! 
Maybe i miss something but i can run everything local if my 'openai/clip-vit-large-patch14'  (only configs without weights) 
folder is where i execute my script, (https://github.com/huggingface/diffusers/pull/5440) but i can't move it anywhere and point script to use it? To be honest idk, why this should work like this.

**Reply by asomoza** [reply, marked as answer]
I wrote why before, maybe it wasn't clear enough, this is something that comes from the [huggingface_hub](https://github.com/huggingface/huggingface_hub) that's used to load the files. If you don't have online access you need to have the same repo name as a local file structure, it's not something that was coded in diffusers.

https://github.com/huggingface/diffusers/blob/f96e4a16adb4c31bab4c0a3d0d145ed2b086ecb0/src/diffusers/loaders/single_file.py#L239

Also I gave you three options you can use to change this, one of the is to use a custom `cache_dir` but you need to either do an online run or copy the files from another system with online access (you can't just copy the files you manually download).

If you're asking if there's and argument you can pass it to load a specific config file for each one of the models inside a single file checkpoint from somewhere in your system, then the answer is no and probably it never will be, due to the added complexity it will introduce for a very specific use case.

**Reply by olegchomp** [reply]
Okey! Thank you!

**Reply by asomoza** [reply]
glad to help
